Framework/DataHandling/src/GroupDetectors2.cpp

```diff
@@ -669,29 +670,25 @@ void GroupDetectors2::processMatrixWorkspace(
 
     // translate to detectors
     std::vector<detid_t> det_ids;
-    if (spectrumInfo.hasDetectors(i)) {
+
+    // If the detector was not found or was not in a group, then ignore it.
+    if (spectrumInfo.spectrumDefinition(i).size() > 1) {
```

Comment:
I did some further cleaning up, and now the `DetectorGroup` include is removed from the algorithm.